### ISSUE TITLE
ArgumentError of cos(Interval(-pi/2, 3pi/2))

### ISSUE THREAD

**Issue by kriukov** [dpsanders/ValidatedNumerics.jl#152]
I am running into a problem with the cos() function. When I try to do

cos(Interval(-pi/2, 3pi/2))

or

cos(Interval(-1.5707963267948966, 4.71238898038469))

I get an ArgumentError:

ERROR: ArgumentError: Must have a <= b to construct Interval(a, b)

(with a, b = 6.123233995736765e-17, -1.8369701987210297e-16).

**Comment by lbenet** [comment]
I understand the problem; I'll try to look onto it tomorrow.

**Comment by dpsanders** [comment]
This is also wrong:
```
julia> sin(Interval(-pi/2, 3pi/2))
[-1, -0.999999]
```

**Comment by dpsanders** [comment]
Fix proposed in #155 . These should return `[-1,1]`, correct?

**Comment by lbenet** [comment]
Just a short remark:

```julia
julia> cos(Interval(-pi/2, 3pi/2))
ERROR: ArgumentError: Must have a ≤ b to construct Interval(a, b).
 in call at /Users/benet/.julia/v0.4/ValidatedNumerics/src/intervals/intervals.jl:19
 in cos at /Users/benet/.julia/v0.4/ValidatedNumerics/src/intervals/trigonometric.jl:86

julia> cos(@interval(-pi/2, 3pi/2))
[-1, 1]
```

The point of this is that the use of `Interval` with -pi/2 and 3pi/2 is not correct to build an Interval that contains *both* -pi/2 and 3pi/2:
```julia
julia> displaymode(format=:full)

julia> Interval(-pi/2, 3pi/2)
Interval(-1.5707963267948966, 4.71238898038469)

julia> @interval(-pi/2, 3pi/2)
Interval(-1.5707963267948968, 4.712388980384691)
```

In any case, something should be corrected because, despite of the issue of using `Interval` or `@interval`, the construction of the result should (automatically) work.

**Comment by dpsanders** [comment]
I agree that `@interval` is the semantically correct thing to use here, but 
`cos(Interval(-pi/2, 3pi/2))` is a valid thing to write (it is an interval between two floating point numbers) and should give a (correct) result!

**Comment by lbenet** [comment]
I agree; the issue about correctly rounding an interval is similar to the same `Interval(0.1)`, which does not contain 0.1. The problem here is that, internally, the implementation of `cos` is producing a lower bound which is greater than the upper bound, which indeed it shouldn't be happening.

**Comment by dpsanders** [comment]
Fixed in #155.
